request-server_test.go

```diff
@@ -345,6 +346,35 @@ func TestRequestFstat(t *testing.T) {
 	checkRequestServerAllocator(t, p)
 }
 
+func TestRequestFsetstat(t *testing.T) {
+	p := clientRequestServerPair(t)
+	defer p.Close()
+	_, err := putTestFile(p.cli, "/foo", "hello")
+	require.NoError(t, err)
+	fp, err := p.cli.OpenFile("/foo", os.O_WRONLY)
+	require.NoError(t, err)
+	err = fp.Truncate(2)
+	require.NoError(t, err)
+	err = fp.Truncate(5)
+	require.NoError(t, err)
+	fi, err := fp.Stat()
+	if assert.NoError(t, err) {
+		assert.Equal(t, fi.Name(), "foo")
+		assert.Equal(t, fi.Size(), int64(5))
```

Comment:
We still want a test of `f.Size() == 2` after the `Truncate(2)`. Imagine an implementation doesn’t do anything with `Truncate` at all, then after `Write("12345"): Truncate(2); Truncate(5)` it would still have a length of `5`, because it never changed the length at all.

Also, use a `require.NoError(t, err)` for both, and don’t embed them in an `if … { … }` block.